### PR TITLE
gdb: enable multiarch by default and split

### PR DESCRIPTION
**Pull request by classabbyamp**
#### Testing the changes
- I tested the changes in this PR: **briefly**

split to prevent `gdb` from being massive

cc @duncaen

### CODE REVIEW

File: srcpkgs/gdb/template [hunk 1]
```diff
@@ -60,3 +65,12 @@ gdb-common_package() {
 		vmove usr/share
 	}
 }
+
+gdb-multiarch_package() {
+	short_desc+=" - all architectures"
+	#depends="${sourcepkg}>=${version}_${revision}"
```
Comment (Duncaen) [inline srcpkgs/gdb/template:71]: This is commented out

File: srcpkgs/gdb/template [hunk 2]
```diff
@@ -60,3 +65,12 @@ gdb-common_package() {
...
+	pkg_install() {
+		vmove usr/bin/run-*
```
Comment (Duncaen) [inline srcpkgs/gdb/template:73]: pattern should be quoted

**Comment by shaohme** [comment]
> #### Testing the changes
> 
>     * I tested the changes in this PR: **briefly**
> 
> 
> split to prevent `gdb` from being massive
> 
> cc @Duncaen

I've tried the branch out. It produces static linked run-* executables in /usr/bin, and seems to work.
Tried gdbserver a simple C program on a debian aarch64 system, and "target remote" from a x86-64 void system. Symbols are read and line stepping works too.

**Comment by classabbyamp** [comment]
disabled it on 32bit as it seems broken

**Comment by shaohme** [comment]
> disabled it on 32bit as it seems broken (link errors for some bpf symbols)

Isn't that what the `archs="i686 x86_64"` is for?

**Comment by classabbyamp** [comment]
no, that would disable the entire gdb package

**Pull request merged**